### PR TITLE
Winduprule 475 spring integration quarkus

### PR DESCRIPTION
**Pull request by jonathanvila**
https://issues.redhat.com/browse/WINDUPRULE-475

### CODE REVIEW

**Comment by PhilipCattanach** [comment]
@jonathanvila - I have reviewed the rules and executed the tests. The rules all fired. They are nicely worded and the issue report links work and are appropriate. Regarding rule # 00000 I think the effort needs to be changed from 10 in keeping with the RHAMT SP documentation. https://access.redhat.com/documentation/en-us/red_hat_application_migration_toolkit/4.3/html/cli_guide/reference_material#how_story_points_are_estimated_in_rules. Thanks

**Comment by jonathanvila** [comment]
> Regarding rule # 00000 I think the effort needs to be changed from 10 in keeping with the RHAMT SP documentation. 
+1 , changed to 7

**Comment by jonathanvila** [comment]
retest

**Comment by PhilipCattanach** [comment]
The rule tested successfully. Only one minor change required. Can the message for springboot-integration-to-quarkus-00000 be changed from "need" to "needs"
Thank you.

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <sourceTechnology id="springboot" />
+        <targetTechnology id="quarkus" />
+    </metadata>
+    <rules>
+        <rule id="springboot-integration-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework.integration" artifactId="spring-integration-core"/>
+            </when>
+            <perform>
+                <hint title="SpringBoot Integration is not supported. A migration to Apache Camel is mandatory." effort="7" category-id="mandatory">
```
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:21]: If it's in the message body then it doesn't need to be in the title too
```suggestion
                <hint title="SpringBoot Integration is not supported" effort="7" category-id="mandatory">
```

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            </perform>
+        </rule>
+        <rule id="springboot-integration-to-quarkus-00010">
+            <when>
+                <xmlfile matches="//*[local-name()='channel']">
+                    <namespace prefix="int" uri="http://www.springframework.org/schema/integration"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <hint title="SpringBoot Integration flow is not supported. A migration to Apache Camel route is mandatory." effort="5" category-id="mandatory">
```
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:38]: If it's in the message body then it doesn't need to be in the title too
```suggestion
                <hint title="SpringBoot Integration flow is not supported" effort="5" category-id="mandatory">
```

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                    <link title="Quarkus Apache Camel" href="https://quarkus.io/guides/camel" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="springboot-integration-to-quarkus-00020">
+            <when>
+                <javaclass references="org.springframework.integration.dsl.IntegrationFlow"/>
+            </when>
+            <perform>
+                <hint title="SpringBoot IntegrationFlow is not supported. A migration to Apache Camel route using From is mandatory." effort="5" category-id="mandatory">
```
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:51]: If it's in the message body then it doesn't need to be in the title too
```suggestion
                <hint title="SpringBoot IntegrationFlow is not supported" effort="5" category-id="mandatory">
```

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 2 above)
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:38]: Is right having effort `5` in this rule (and the next one) and effort `7` in the first rule?
Are they covering different scenarios that justify different effort?

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    <message>
+                        SpringBoot Integration flow is not supported. A migration to Apache Camel route is mandatory.
+                    </message>
+                    <link title="Quarkus Apache Camel" href="https://quarkus.io/guides/camel" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="springboot-integration-to-quarkus-00020">
+            <when>
+                <javaclass references="org.springframework.integration.dsl.IntegrationFlow"/>
```
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:48]: Is it worth adding a location? `import`?

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                        Spring dependency `org.springframework.boot:spring-boot-starter-integration` need to move to `org.apache.camel.quarkus:camel-quarkus-parent`.
+                        Spring Integration Flows must be recoded into Apache Camel Routes.
+                    </message>
+                    <link title="Quarkus Apache Camel" href="https://quarkus.io/guides/camel" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="springboot-integration-to-quarkus-00010">
+            <when>
+                <xmlfile matches="//*[local-name()='channel']">
```
Comment (mrizzi) [inline rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:33]: Enhanced in https://github.com/jonathanvila/windup-rulesets/pull/1

**Comment by jonathanvila** [comment]
> Other than the comments, this PR is missing the technology usage rules that #429 has.

Spring Integration was already tackled by an old rule : https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/technology-usage/integration.windup.xml#L45

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 1 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:21]: ok

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 2 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:38]: ok

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 3 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:51]: ok

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 2 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:38]: Well, the first one implies the whole dependency , these others only the specific usage.

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 4 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml:48]: ok

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    <rules>
+        <rule id="springboot-integration-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework.integration" artifactId="spring-integration-core"/>
+            </when>
+            <perform>
+                <hint title="SpringBoot Integration is not supported. A migration to Apache Camel is mandatory." effort="7" category-id="mandatory">
+                    <message>
+                        SpringBoot Integration is not supported. A migration to Apache Camel is mandatory.
+                        Spring dependency `org.springframework.boot:spring-boot-starter-integration` need to move to `org.apache.camel.quarkus:camel-quarkus-parent`.
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml]: +1

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 5 above)
Comment (jonathanvila) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml]: done

File: rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml (hunk 2 above)
Comment (mrizzi) [reply rules-reviewed/quarkus/springboot/springboot-integration-to-quarkus.windup.xml]: Fair enough

**Comment by jonathanvila** [comment]
> Discussed with @PhilipCattanach the titles of the hints (see screenshot)
> Screenshot from 2020-04-28 13-16-10
> 
> It's better to have consistency so please change `Integration flow` to become `IntegrationFlow`

In fact that's on purpose..... IntegrationFlow is a class, Integration flow is the concept
@PhilipCattanach @mrizzi are you still wanting me to change it ?

**Pull request merged**